File: linked_list/linked_list.py
class LinkedList(object):
    def __init__(self, data):
        self.data = data
        self.next = None

    def search(self, target):
        if self.data == target:
            return self

        if not self.next:
            return None

        return self.next.search(target)     

if __name__ == "__main__":
    doesNotContainFish = LinkedList("Dog")
    print("search fails: {}".format(doesNotContainFish.search("Fish")))

    containsFish = LinkedList("Fish")
    print("search succeeds: {}".format(containsFish.search("Fish")))

File: linked_list/LinkedList.swift
class LinkedList {
    var data: String
    var next: LinkedList?

    init(data: String) {
        self.data = data
        self.next = nil
    }

    func search(target: String) -> LinkedList? {        
        if (data != nil && data == target) {
            return self
        } 

        if (next == nil) {
            return nil
        }
        
        return next?.search(target: target)
    }
}

var doesNotContainFish = LinkedList(data: "Dog")
print("search fails: ")
print(doesNotContainFish.search(target: "Fish")) 

var containsFish = LinkedList(data: "Fish")
print("search succeeds: ")
print(containsFish.search(target: "Fish"))